lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -1019,9 +1019,8 @@ void ExportVerilog::prepareHWModule(hw::HWModuleOp module,
 
 namespace {
 
-struct TestPrepareForEmissionPass
-    : public TestPrepareForEmissionBase<TestPrepareForEmissionPass> {
-  TestPrepareForEmissionPass() {}
+struct PrepareForEmissionPass
+    : public PrepareForEmissionBase<PrepareForEmissionPass> {
   void runOnOperation() override {
     HWModuleOp module = getOperation();
     LoweringOptions options = getLoweringCLIOption(
```

Comment:
This means the options are taken from the CLI options only, not the module, is that correct?

Would be good to audit the plumbing of these options so we don't have any surprises, oof.

I think one result of this change (I think?) is any LoweringOptions on the module going into ExportVerilog will be ignored for this portion of the lowering (only using the CLI options), but not for the bits that grab them from the module (`LoweringOptions options(module)`)?  We do a `applyLoweringCLOptions` which IIRC merges them, but here this runs prepare only with the CLI options.  Generally I'd expect these to be the same, but they may not be-- for example, when saving a reproducer for example it's nice to not have to specify them if they're already on the module.

Mostly I'm concerned about ways that different flows might end up with /different/ options at various points.

Reply:
> This means the options are taken from the CLI options only, not the module, is that correct?

This `getLoweringCLIOption` tries getting a lowering option from CLI option first, and then checks module's attribute. (Sorry for naming a confusing method, it should be renamed for more clear name)

Reply:
You make me rethink all the lowering options.

Reply:
Just making sure I got this right:  I think the original plan was that the command line options would be set on the module (by `firtool`), and then all passes would read the options from there. I guess that missed the use-case of `circt-opt` wanting to take advantage of the CLI options, and suddenly we have passes reading CLI options directly, which we don't think is a great idea.  We are moving `firtool` back to the original design, but adding a new utility pass to `circt-opt` to apply command line options to the module.  Due to the main functionality of `circt-opt` living in MLIR core, we don't have the option to add the same command line options there.

Reply:
yes, this makes a separate pass for testing which can be run from circt-opt to set options on a module.  firtool was doing it before, but it was also being reapplied by exportverilog.